src/lamplib/src/genny/tasks/run_tests.py

```diff
@@ -69,29 +71,44 @@ def _run_command_with_sentinel_report(
         )
 
 
-def cmake_test(genny_repo_root: str, workspace_root: str):
+def cmake_test(
+    genny_repo_root: str, workspace_root: str, regex: str = None, repeat_until_fail: int = 1
+):
     info = toolchain.toolchain_info(genny_repo_root=genny_repo_root, workspace_root=workspace_root)
     workdir = os.path.join(genny_repo_root, "build")
 
     xunit_dir = os.path.join(workspace_root, "build", "XUnitXML")
     os.makedirs(xunit_dir, exist_ok=True)
 
+    print("repeat until fail: ", str(repeat_until_fail))
     ctest_cmd = [
         "ctest",
-        "--verbose",
+        "--schedule-random",
+        "--output-on-failure",
+        "--parallel",
+        "4",
+        "--repeat-until-fail",
+        str(repeat_until_fail),
         "--label-exclude",
         "(standalone|sharded|single_node_replset|three_node_replset|benchmark)",
     ]
 
+    if regex is not None:
+        ctest_cmd += ["--tests-regex", regex]
```

Comment:
do we need to `shlex.quote` or whatever? I forget how argv/quoting is handled downstream. Just double-checking.

Reply:
Handled downstream, fortunately!